File: FinancialMonitor.Api/Entities/Transaction.cs
using FinancialMonitor.Api.Enums;

namespace FinancialMonitor.Api.Entities;

public class Transaction
{
    public Guid TransactionId { get; set; }

    public decimal Amount { get; set; }

    public Currency Currency { get; set; }

    public TransactionStatus Status { get; set; }

    public DateTime Timestamp { get; set; }
}

File: FinancialMonitor.Client/src/models/transaction.ts
import type { Currency } from "../enums/currency";
import type { TransactionStatus } from "../enums/transactionStatus";

export interface Transaction {
    transactionId: string;
    amount: number;
    currency: Currency;
    status: TransactionStatus;
    timestamp: string;
}

export interface CreateTransactionRequest {
    amount: number;
    currency: Currency;
}